testing/README.md

```diff
@@ -0,0 +1,15 @@
+# Testing Directory README
+
+All files and configurations used for testing as detailed in the [Testing Guide](https://karpenter.sh/preview/testing-guide/) live here. Configurations and setup files here will be used in the future to test PRs.
+
+## File Directory
+- `/setup`: Testing infrastructure
+- `/tekton`: Tekton YAMLs defining test suites
```

Comment:
This structure leaks the implementation (i.e. tekton) into the folder structure. What if we move to prow? Do we need to rename all of these folders? 

Consider something like:
`/test/suitename`
`/test/soak`
`/test/performance`